subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/DefaultXcodeProject.java

```diff
@@ -16,30 +16,53 @@
 
 package org.gradle.ide.xcode.internal;
 
+import org.gradle.api.file.FileCollection;
 import org.gradle.ide.xcode.XcodeProject;
+import org.gradle.ide.xcode.internal.xcodeproj.GidGenerator;
 
 import java.io.File;
 import java.util.ArrayList;
-import java.util.HashSet;
+import java.util.Collections;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
 
 public class DefaultXcodeProject implements XcodeProject {
-    private final List<XcodeTarget> targets = new ArrayList<XcodeTarget>();
-    private final List<XcodeScheme> schemes = new ArrayList<XcodeScheme>();
-    private final Set<File> sources = new HashSet<File>();
+    private final GidGenerator gidGenerator = new GidGenerator(Collections.<String>emptySet());
```

Comment:
Each Gradle project generates its own XCode project. They are somehow independent so if there is a collision, it won't affect the other project. Using the `GidGenerator` for the whole workspace would mean we need to share it between Gradle project instance. I'm not sure we want to get into this given the internal list isn't synchronized.